tests/test_sshfsmount.cpp

```diff
@@ -299,57 +299,71 @@ TEST_P(SshfsMountExecuteThrowRuntErr, test_runtime_error_when_executing)
 
 // Commands to check that an unknown version of FUSE gives a correct answer.
 CommandVector unk_fuse_cmds = {{"sudo env LD_LIBRARY_PATH=/foo/bar /baz/bin/sshfs -V", "weird fuse version\n"},
                                {"sudo env LD_LIBRARY_PATH=/foo/bar /baz/bin/sshfs -o slave -o transform_symlinks -o "
-                                "allow_other :\"source\" \"target\"",
+                                "allow_other :\"source\" \"/home/ubuntu/target\"",
                                 "don't care\n"}};
 
 // Commands to check that the server correctly creates the mount target.
 CommandVector exec_cmds = {
-    {"sudo /bin/bash -c 'P=\"/home/ubuntu/target\"; while [ ! -d \"$P/\" ]; do P=${P%/*}; done; echo $P/'",
-     "/home/ubuntu/\n"},
+    {"sudo /bin/bash -c 'P=/home/ubuntu/target; while [ ! -d $P/ ]; do P=${P%/*}; done; echo $P/'", "/home/ubuntu/\n"},
+    {"sudo /bin/bash -c 'cd \"/home/ubuntu/\" && mkdir -p \"target\"'", "\n"},
+    {"sudo /bin/bash -c 'cd \"/home/ubuntu/\" && chown -R 1000:1000 \"target\"'", "\n"}};
+
+// Commands to check that the ~ expansion works.
+CommandVector tilde1_cmds = {
+    {"echo ~/target", "/home/ubuntu/target\n"},
+    {"sudo /bin/bash -c 'P=/home/ubuntu/target; while [ ! -d $P/ ]; do P=${P%/*}; done; echo $P/'", "/home/ubuntu/\n"},
+    {"sudo /bin/bash -c 'cd \"/home/ubuntu/\" && mkdir -p \"target\"'", "\n"},
+    {"sudo /bin/bash -c 'cd \"/home/ubuntu/\" && chown -R 1000:1000 \"target\"'", "\n"}};
+
+// Commands to check that the ~user expansion works (assuming that user exists).
+CommandVector tilde2_cmds = {
+    {"echo ~ubuntu/target", "/home/ubuntu/target\n"},
+    {"sudo /bin/bash -c 'P=/home/ubuntu/target; while [ ! -d $P/ ]; do P=${P%/*}; done; echo $P/'", "/home/ubuntu/\n"},
     {"sudo /bin/bash -c 'cd \"/home/ubuntu/\" && mkdir -p \"target\"'", "\n"},
     {"sudo /bin/bash -c 'cd \"/home/ubuntu/\" && chown -R 1000:1000 \"target\"'", "\n"}};
 
 // Commands to check that the server works if an absolute path is given.
 CommandVector absolute_cmds = {
-    {"sudo /bin/bash -c 'P=\"/home/ubuntu/target\"; while [ ! -d \"$P/\" ]; do P=${P%/*}; done; echo $P/'",
-     "/home/ubuntu/\n"},
+    {"sudo /bin/bash -c 'P=/home/ubuntu/target; while [ ! -d $P/ ]; do P=${P%/*}; done; echo $P/'", "/home/ubuntu/\n"},
     {"sudo /bin/bash -c 'cd \"/home/ubuntu/\" && mkdir -p \"target\"'", "\n"},
     {"sudo /bin/bash -c 'cd \"/home/ubuntu/\" && chown -R 1000:1000 \"target\"'", "\n"}};
 
 // Commands to check that it works for a nonexisting path.
 CommandVector nonexisting_path_cmds = {
-    {"sudo /bin/bash -c 'P=\"/nonexisting/path\"; while [ ! -d \"$P/\" ]; do P=${P%/*}; done; echo $P/'", "/\n"},
+    {"sudo /bin/bash -c 'P=/nonexisting/path; while [ ! -d $P/ ]; do P=${P%/*}; done; echo $P/'", "/\n"},
     {"sudo /bin/bash -c 'cd \"/\" && mkdir -p \"nonexisting/path\"'", "\n"},
     {"sudo /bin/bash -c 'cd \"/\" && chown -R 1000:1000 \"nonexisting\"'", "\n"}};
```

Comment:
You don't need to have the `mkdir` and `chown` lines everywhere - it's probably enough that just one test verifies those, and the rest work with the defaults.

What you're not testing for, though, is the escaping. We need at least one test case that verifies the escaping is done properly.

Reply:
OK, I don't have a strong opinion here. I think testing those doesn't hurt, but you're right in the sense that these lines are almost trivial and will work iff the first one does.

I left the first `mkdir` and `chown`'s.